### PR TITLE
chore: Make asciidoc sole primary documentation source, deleting .in files

### PR DESCRIPTION
**Pull request by gwhitney**
Currently, the Fluxbox repository contains both asciidoc files from which man pages can be generated, and the resulting manpages generated from them (except for a final substitution of the destination directory for the package). This is a maintenance problem for at least two main reasons:

1) When a developer changes the asciidoc and regenerates the man pages, they may contain additional spurious changes such as the version number of the asciidoc software used, polluting the git history of the files.

2) Some developers may not be aware of the asciidoc versions of the files and check changes directly in to the man pages, which are then lost when the man pages are later regenerated from the asciidoc, and must be restored (cf the very recent commit 1d19662).

In short, having both the asciidoc and the derived files in the git repository violates the "Don't Repeat Yourself" principle, with all of the associated drawbacks.

This PR eliminates the manpage format files from the repository, making the asciidoc the sole, primary documentation source. It incorporates the asciidoc generation step into the overall project make. This of course requires that all developers install asciidoc and xmlto, but frankly, that de facto should already have been the case as the only way that the manpages should have been updated was by editing the asciidoc and then re-processing them (but this procedure was not very prominently advertised - it required looking down into the doc/asciidoc subdirectory and noticing the README there). Now, of course, it will be clearly necessary to edit the `doc/asciidoc/*.txt` files to update the documentation.

### CODE REVIEW

**Comment by mgumz** [comment]
the reason to have both the asciidoc files as well as the generated files in the repo is this: it removes the compile time requirement to asciidoc and mainly its depencies. try with an empty system and "apt install asciidoc" to see what i mean.

the situation, that someone is really working on the documentation is … once in a couple of years.

on the other hand, the main way for people to get fluxbox compiled is by unpacking the src.tar.gz … and there they expect a fully functional `configure` script as well. on a git-checkout, that `configure` script is generated by tools which do not have to be on the machine where the code is compiled.

so, i think i am ok with the new approach but then the generated files have to be included in the src.tar.gz when making a release.

**Comment by gwhitney** [comment]
Which src.tar.gz? Do you mean the `fluxbox-V.V.V.tar.gz` file that is created by `make dist`? If so, I just tried that on this exact PR (rebased) and this resulting tarfile does include all of the .1 and .5 man files and .1.in and .5.in nroff files (but not the asciidoc .txt files). So is there any reason this PR among the various ones I submitted isn't ready for merging as it is? If so, please let me know and I will be happy to do whatever is needed. Thanks.

**Comment by mgumz** [comment]
@gwhitney let me rephrase: a developer is using git to checkout the fluxbox-src code and everything is there - the asciidoc files as well.

when a user gets the fluxbox-v.v.v.tar.gz file, she usually is not interested in providing patches against the documentation but rather wants to `make install` and thats it. and I, personally, dislike to have to install something like `asciidoctor` just to compile src-code to get a binary for my platform.

so, one could also tackle the issue to get rid of .in etc in the the git-repository by adding steps in the release pipeline to create those files before packaging fluxbox-v.v.v.tar.gz i think

**Comment by mgumz** [comment]
> So is there any reason this PR among the various ones I submitted isn't ready for merging as it is?

Lack of time to properly check out / test the functional patches you provided. It is not by bad intention.

**Comment by gwhitney** [comment]
I wasn't questioning any developer's intention or anything. I am simply looking for guidance as to what if anything I can do to move the merging of this PR forward. As far as I can tell, it does in fact have the property suggested: starting from a fresh clone of the repository, executing `make dist` (presuming asciidoc is installed) produces a `fluxbox-v.v.v.tar.gz` file with the generated .1 and .5 man files. Please let me know if there is anything else I can do to help with this. Thanks.

**Comment by mgumz** [comment]
@gwhitney I have checked `make dist` … seems to work (generates the docs).

still, i somewhat dislike the requirement to have asciidoc around when i do a `git checkout` and `make` and `make install`. 
i see your point in regarding it … might confuse people wanting to edit the documentation and then edit the wrong files. your README helps, maybe putting the generated files into a .docs folder but still have it in the repository. what do you think?
